modules/dnn/test/test_model.cpp

```diff
@@ -202,6 +221,22 @@ TEST_P(Test_Model, DetectionMobilenetSSD)
                     scoreDiff, iouDiff, confThreshold, nmsThreshold, size, mean, scale);
 }
 
+TEST_P(Test_Model, Segmentation)
+{
+    std::string inp = _tf("dog416.png");
+    std::string weights_file = _tf("fcn8s-heavy-pascal.prototxt");
+    std::string config_file = _tf("fcn8s-heavy-pascal.caffemodel");
+    std::string exp = _tf("segmentation_exp.png");
+
+    Size size{512, 512};
+    float norm = 0;
+    double scale = 1.0;
+    Scalar mean = Scalar(127.5, 127.5, 127.5);
```

Comment:
Is this correct mean values? According to [models.yml](https://github.com/opencv/opencv/blob/master/samples/dnn/models.yml), there is no mean subtraction for FCN8s network and no Red<->Blue channels swap.

Reply:
I guess, I can change it, but does it really matter for test?

Reply:
@dvd42, It's very important. Tests as samples are used by developers as reference code. If we'll provide wrong usage in it - we'll mislead all the users.

Reply:
So the question is how you generated test data? You just run it once and collected output? Then how you can proof that the changes are correct?

Please use a sample to generate reference data for specific input size and input image: https://github.com/opencv/opencv/blob/master/samples/dnn/segmentation.py

Reply:
@dkurt Every time I make a change I generate new test data

Reply:
@dvd42, By sample?

Reply:
@dkurt What do you mean by sample?

Reply:
Run the following sample to get reference output data: https://github.com/opencv/opencv/blob/master/samples/dnn/segmentation.py:

```
python segmentation.py fcn8s --input /path/to/image.png --width xxx --height xxx
```

Reply:
@dkurt I just ran it and the output is **exactly** the same as my reference data. I dont see the problem.

Reply:
Please reduce input resolution as well. Model is pretty heavy. I think 128x128 would be enough (just check the segmentation mask).

Reply:
@dkurt, done